test/app.test.js

```diff
@@ -0,0 +1,134 @@
+var vumigo = require('vumigo_v02');
+var fixtures = require('./fixtures');
+var AppTester = vumigo.AppTester;
+var test_config = require('./config');
+var utils = require('../src/utils');
+require('should');
+
+describe("app", function () {
+  describe("GoApp", function () {
+    var app;
+    var tester;
+
+    beforeEach(function () {
+      app = new go.app.GoApp();
+
+      tester = new AppTester(app);
+
+      tester
+        .setup.config.app(test_config.config)
+        .setup(function (api) {
+          fixtures().forEach(api.http.fixtures.add);
+        });
+    });
+
+    describe("when the user starts a session", function () {
+      it("should respond with welcome_message", function () {
+        return tester
+          .start()
+          .input('hi')
+          .check.interaction({
+            state: 'states:start',
+            reply: test_config.config.welcome_message
+          })
+          .check.reply.ends_session()
+          .run();
+      });
+    });
+
+    describe("when the user starts a session with a keyword in the message", function () {
+      it("should respond with welcome_message and send an email", function () {
+        return tester
+          .start()
+          .input('test this sms out')
+          .check.interaction({
+            state: 'states:start',
+            reply: test_config.config.welcome_message
+          })
+          .check.reply.ends_session()
+          .run();
+      });
+    });
+
+    describe('utils', function () {
```

Comment:
Ideally the tests for the code in `utils.js` should be in a separate file to the ones for `app.js`.